src/management-system-v2/components/modeler-share-modal.tsx

```diff
@@ -0,0 +1,146 @@
+
+  const handleClose = () => {
+    setIsOpen(false);
+    setIsSharePublicLinkSelected(false);
+    setIsEmbedInWebsiteSelected(false);
+    setActiveIndex(null);
+  };
+
+  const handleCopyXMLToClipboard = async () => {
+    if (modeler) {
+      const { xml } = await modeler.saveXML({ format: true });
+      if (xml) {
+        navigator.clipboard.writeText(xml);
+        message.success('Copied to clipboard');
+      }
+    }
+  };
+
+  const handleOptionClick = (index: number) => {
+    setActiveIndex(index);
+    setIsSharePublicLinkSelected(index === 0);
+    setIsEmbedInWebsiteSelected(index === 1);
+  };
+
+  const options = [
+    {
+      optionIcon: <LinkOutlined style={{ fontSize: '24px' }} />,
+      optionName: 'Share Public Link',
+      optionOnClick: () => handleOptionClick(0),
+    },
+    {
+      optionIcon: (
+        <span>
+          <LeftOutlined style={{ fontSize: '24px' }} />
+          <RightOutlined style={{ fontSize: '24px' }} />
+        </span>
+      ),
+      optionName: 'Embed in Website',
+      optionOnClick: () => handleOptionClick(1),
+    },
+    {
+      optionIcon: <CopyOutlined style={{ fontSize: '24px' }} />,
+      optionName: 'Copy Diagram to Clipboard (PNG)',
+      optionOnClick: () => {
+        handleOptionClick(2);
+        copyProcessImage(modeler);
+      },
+    },
+    {
+      optionIcon: <CopyOutlined style={{ fontSize: '24px' }} />,
+      optionName: 'Copy BPMN to Clipboard (XML)',
+      optionOnClick: () => {
+        handleOptionClick(3);
+        handleCopyXMLToClipboard();
+      },
+    },
+    {
+      optionIcon: <ExportOutlined style={{ fontSize: '24px' }} />,
+      optionName: 'Export as file',
+      optionOnClick: () => {
+        handleOptionClick(4);
+        onExport();
+      },
+    },
+  ];
+
+  return (
+    <>
+      <Modal
+        title={<div style={{ textAlign: 'center' }}>Share</div>}
+        open={isOpen}
+        width={800}
+        closeIcon={false}
+        onCancel={handleClose}
+        footer={
+          <Button onClick={handleClose} style={{ border: '1px solid black' }}>
+            Close
+          </Button>
+        }
+      >
+        <Space
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            flexWrap: 'wrap',
+            justifyContent: 'space-evenly',
+          }}
+        >
+          {options.map((option, index) => (
+            <ModelerShareModalOption
+              key={index}
+              optionIcon={option.optionIcon}
+              optionName={option.optionName}
+              optionOnClick={option.optionOnClick}
+              isActive={index === activeIndex}
+            />
+          ))}
+        </Space>
+        {isSharePublicLinkSelected ? (
+          <>
+            <Divider style={{ backgroundColor: '#000' }} />
+            <ModelerShareModalOptionPublicLink />
+          </>
+        ) : null}
+        {isEmbedInWebsiteSelected ? (
```

Comment:
Unnecessary code duplication. See above

Reply:
updated code pushed to the m2/modeler-share-modal branch